### PR TITLE
Improve gce plugin

### PR DESCRIPTION
**Pull request by astj**
ref: #304

Thanks to littlekbt, but we need some improvements.
- Make `-project` and `-instance-name` specifiable from command line params
- Improve error handling and logging

### CODE REVIEW

File: mackerel-plugin-gcp-compute-engine/README.md [hunk 1]
```diff
@@ -7,23 +7,25 @@ mackerel-plugin-gcp-compute-engine is mackerel plugin that collects metrics from
...
+Caution: This Plugin works only on Compute Engien instance that is enabled Stackdriver Monitoring API full access.
 
 ## Usage
 
 ```shell
-mackerel-plugin-gcp-compute-engine [-api-key=<APIKey>]
+mackerel-plugin-gcp-compute-engine -api-key=<APIKey> [-project=<project_id_or_number>] [-instance-name=<target instance name>]
 ```
 
+If `-project` or `-instance-name` are not specified, they are obtained from Google Compiute Engine Metadata API for the instance executing this plugin.
```
Comment (stefafafan) [inline mackerel-plugin-gcp-compute-engine/README.md]: [nit] s/Compiute/Compute

File: mackerel-plugin-gcp-compute-engine/README.md [hunk 2]
```diff
@@ -7,23 +7,25 @@ mackerel-plugin-gcp-compute-engine is mackerel plugin that collects metrics from
...
 
 
-Caution: This Plugin works only on Compute Engien instance that is enabled Stackdriver Monitoring API full access. 
+Caution: This Plugin works only on Compute Engien instance that is enabled Stackdriver Monitoring API full access.
 
 ## Usage
 
 ```shell
-mackerel-plugin-gcp-compute-engine [-api-key=<APIKey>]
+mackerel-plugin-gcp-compute-engine -api-key=<APIKey> [-project=<project_id_or_number>] [-instance-name=<target instance name>]
```
Comment (stefafafan) [inline mackerel-plugin-gcp-compute-engine/README.md]: [nit] The format (e.g. capitalization) of `APIKey` , `project_id_or_number` , and `target instance name` (and `YOUR-API-KEY` below this) seem inconsistent.

File: mackerel-plugin-gcp-compute-engine/lib/gcp-compute-engine.go [hunk 3]
```diff
@@ -211,33 +214,42 @@ func getInstanceName() string {
 
 // Do the plugin
 func Do() {
+	optProject := flag.String("project", "", "Project No")
```
Comment (stefafafan) [inline mackerel-plugin-gcp-compute-engine/lib/gcp-compute-engine.go]: [nit] `Project No` seemed a bit vague to me, what is `No` ?

**Comment by astj** [comment]
@stefafafan Thank you for your review. fixed!

**Comment by stefafafan** [comment]
@astj Thanks!
Looks good to merge to me 👍

**Pull request merged**